xarray/plot/utils.py

```diff
@@ -253,12 +254,67 @@ def _determine_cmap_params(plot_data, vmin=None, vmax=None, cmap=None,
                 levels=levels, norm=norm)
 
 
-def _infer_xy_labels(darray, x, y):
+def _infer_xy_labels_3d(darray, x, y, rgb):
+    """
+    Determine x and y labels for showing RGB images.
+
+    Attempts to infer which dimension is RGB/RGBA by size and order of dims.
+
+    """
+    # Start by detecting and reporting invalid combinations of arguments
+    assert darray.ndim == 3
+    not_none = [a for a in (x, y, rgb) if a is not None]
+    if len(set(not_none)) < len(not_none):
+        raise ValueError('Dimensions passed as x, y, and rgb must be unique.')
+    for label in not_none:
+        if label not in darray.dims:
+            raise ValueError('%r is not a dimension' % (label,))
+
+    # Then calculate rgb dimension if certain and check validity
+    could_be_color = [label for label in darray.dims
+                      if darray[label].size in (3, 4) and label not in (x, y)]
+    if rgb is None and not could_be_color:
+        raise ValueError(
+            'A 3-dimensional array was passed to imshow(), but there is no '
+            'dimension that could be color.  At least one dimension must be '
+            'of size 3 (RGB) or 4 (RGBA), and not given as x or y.')
+    if rgb is None and len(could_be_color) == 1:
+        rgb = could_be_color[0]
+    if rgb is not None and darray[rgb].size not in (3, 4):
+        raise ValueError('Cannot interpret dim %r of size %s as RGB or RGBA.'
+                         % (rgb, darray[rgb].size))
+
+    # If rgb dimension is still unknown, there must be two or three dimensions
+    # in could_be_color.  We therefore warn, and use a heuristic to break ties.
+    if rgb is None:
+        assert len(could_be_color) in (2, 3)
+        if darray.dims[-1] in could_be_color:
+            rgb = darray.dims[-1]
+            warnings.warn(
+                'Several dimensions of this array could be colors.  Xarray '
+                'will use the last dimension (%r) to match '
+                'matplotlib.pyplot.imshow.  You can pass names of x, y, '
+                'and/or rgb dimensions to override this guess.' % rgb)
+        else:
+            rgb = darray.dims[0]
+            warnings.warn(
+                '%r has been selected as the color dimension, but %r would '
+                'also be valid.  Pass names of x, y, and/or rgb dimensions to '
+                'override this guess.' % darray.dims[:2])
+    assert rgb is not None
+
+    # Finally, we pick out the red slice and delegate to the 2D version:
+    return _infer_xy_labels(darray.isel(**{rgb: 0}).squeeze(), x, y)
+
+
+def _infer_xy_labels(darray, x, y, imshow=False, rgb=None):
     """
     Determine x and y labels. For use in _plot2d
 
-    darray must be a 2 dimensional data array.
+    darray must be a 2 dimensional data array, or 3d for imshow only.
     """
+    if imshow and darray.ndim == 3:
+        return _infer_xy_labels_3d(darray, x, y, rgb)
```

Comment:
Please make sure this raises an informative error message if RGB is provided and the input has the wrong number of dimensions or isn't being plotted with imshow.